source/fundamentals/object-serialization.txt

```diff
@@ -0,0 +1,242 @@
+is called deserialization. MongoDB stores data as documents containing
+fields with BSON values. The driver uses object
+serialization to translate data between BSON documents and Java objects.
+You can specify how the driver should perform serialization
+using **codecs**.
+
+Codecs implement translation between Extended JSON and Java objects
+according to the ``Codec`` interface using methods that serialize and
+deserialize data. A ``CodecRegistry`` is a class that contains a list
+of codec instances. You can specify a ``CodecRegistry`` for a
+``MongoClient``, ``MongoDatabase``, or ``MongoCollection`` to be
+used for all subsequent operations on that object.
+Whenever you send or receive data from a connected MongoDB instance,
+the driver selects the first codec that can translate each object in the
+``CodecRegistry`` of the ``MongoClient``, ``MongoDatabase``, or
+``MongoCollection`` used to communicate with the MongoDB instance.
+
+.. _default-codecs:
+
+Default Codecs
+--------------
+
+By default, every ``MongoClient`` uses a ``CodecRegistry`` that contains
+codecs allowing the serialization of three classes and many standard
+Java types those classes can contain, including objects of type
+``Integer``, ``Long``, ``Date``, and more:
+
+.. list-table:: Default Serializable Classes
+   :widths: 40 60
+   :stub-columns: 1
+   :header-rows: 1
+
+   * - Class
+     - Codec
+
+   * - :java-docs:`Document <apidocs/bson/org/bson/Document.html>`
+     - :java-docs:`DocumentCodec <apidocs/bson/org/bson/codecs/DocumentCodec.html>`
+
+   * - :java-docs:`BasicDBObject <apidocs/mongodb-driver-core/com/mongodb/BasicDBObject.html>`
+     - :java-docs:`DBObjectCodec <apidocs/mongodb-driver-core/com/mongodb/DBObjectCodec.html>`
+
+   * - :java-docs:`BsonDocument <apidocs/bson/org/bson/BsonDocument.html>`
+     - :java-docs:`BsonDocumentCodec <apidocs/bson/org/bson/codecs/BsonDocumentCodec.html>`
+
+Because the default ``CodecRegistry`` contains the codecs for these
+classes, you can use the ``Document``, ``BasicDBObject``, and
+``BsonDocument`` classes to represent data in a ``MongoCollection``
+without explicitly specifying a codec. For instance, the following
+example creates three instances of the ``exampleCollection`` collection
+in the ``exampleDatabase`` database, one for each class with
+serialization support in the default ``CodecRegistry``:
+
+.. code-block:: java
+
+   MongoClient client = MongoClients.create("mongodb+srv://<user>:<password>@<cluster-url>");
+   MongoDatabase database = client.getDatabase("exampleDatabase");
+   MongoCollection<Document> documentCollection =
+       database.getCollection("exampleCollection", Document.class);
+   MongoCollection<BasicDBObject> basicDBObjectCollection =
+       database.getCollection("exampleCollection", BasicDBObject.class);
+   MongoCollection<BsonDocument> BsonDocumentCollection =
+       database.getCollection("exampleCollection", BsonDocument.class);
+
+Custom Codecs
```

Comment:
We should mention the POJO codec or link off to it before going through custom codecs. 

My thought process is that with the default registry and the PojoCodec facility most users shouldn't have to need to create their own codecs.

Reply:
Unfortunately we haven't created the documentation for POJO codecs yet. I was thinking that we could link over to it once that page has been written.

Reply:
Sounds good - perhaps when POJO codecs are documented removing custom codecs to their own page would be good.  So the flow would be Object Serialization -> PojoCodecs -> Custom Codecs.

Reply:
This can be resolved